src/Firebase/FirebaseAuth.js

```diff
@@ -0,0 +1,155 @@
+import { app } from './FirebaseConfig';
+import {
+	getAuth,
+	signOut,
+	GoogleAuthProvider,
+	createUserWithEmailAndPassword,
+	updateProfile,
+	signInWithEmailAndPassword,
+	signInWithPopup,
+} from 'firebase/auth';
+import { addUserToTheDB, getSpecificUser } from './FirebaseFirestore';
+const auth = getAuth(app);
+const defaultImage =
+	'https://firebasestorage.googleapis.com/v0/b/react-testing-firebase-39d5e.appspot.com/o/defaultImages%2Fdefault.jpg?alt=media&token=1f3c0b1d-9d99-406b-b524-f6e278f7c5c7';
+
+const getNewUserObject = (currentUser) => {
+	return {
+		displayName: currentUser.displayName,
+		uid: currentUser.uid,
+		photoURL: currentUser.photoURL,
+		email: currentUser.email,
+		followers: 0,
+		following: 0,
+		posts: 0,
+		bio: '',
+		url: '',
+		backgroundImageUrl: '',
+	};
+};
+const emailSignUpHandler = async (errorHandler, setIsLoading, dispatch, setTextFields, textFields, setAuthUser) => {
+	const { email, password, confirmPassword, name } = textFields;
+	if (name.length < 2) {
+		setTextFields({ ...textFields, nameError: true });
+		errorHandler(true, 'Name must be atleast 2 characters long');
+		return;
+	}
+	if (email.length === 0) {
+		setTextFields({ ...textFields, emailError: true });
+		errorHandler(true, 'Email is required');
+		return;
+	}
+	if (
+		!email
+			.toLowerCase()
+			.match(
+				/^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
+			)
+	) {
+		errorHandler(true, 'Email is invalid');
+		setTextFields({ ...textFields, emailError: true });
+		return;
+	}
+	if (password.length < 6) {
+		errorHandler(true, 'Password cannot be shorter than 6 characters');
+		setTextFields({ ...textFields, passWordError: true });
+		return;
+	}
+	if (confirmPassword.length < 6) {
```

Comment:
It also can be done in the input tag itself just by using minlength attribute 
```suggestion
	if (confirmPassword.length < 6) {
```